core/services/job/job_orm_test.go

```diff
@@ -881,18 +848,37 @@ func Test_FindPipelineRunIDsByJobID(t *testing.T) {
 	for i, j := 0, 0; i < 2500; i++ {
 		mustInsertPipelineRun(t, pipelineORM, jobs[j])
 		j++
-		if j == len(jobs) {
+		if j == len(jobs)-1 {
 			j = 0
 		}
 	}
 
+	// Creation of job runs above cannot run in parallel, otherwise run ids are unpredictable
+	t.Parallel()
```

Comment:
This is clever, although I wonder if there are unexpected consequences... :thinking: 
Is there a significant slowdown in testing the package as a whole if this is just omitted and run serially instead?

Reply:
There shouldn't be unexpected consequences, because each test creates its own jobs, which are queried by job id.  There are already a bunch that have been running in parallel for a long time.  The thing which was unique about this one, is that the pipeline run id's in the table must be sequential.  Usually that doesn't matter, but for testing the batching it matters.

Reply:
I verified locally that if you run go test on the whole directory in parallel with a high -count, it always passes as is, but was consistently failing before.

Reply:
I just mean that I'm unsure of the fundamental implications of calling `t.Parallel()` this late. I have only ever seen it called as the first line. :shrug:

Reply:
My understanding is that the caller of each test is required to block until either `t.Parallel()` is called or it returns.  So `t.Parallel()` is basically just a way to signal the parent "you're unblocked now, it's okay to go ahead and run other tests now, don't wait on me".  (Although it will still wait on the child test again before exiting.)

Reply:
The docs are a bit vague I guess, but at least in the case of explicitly calling `Run()`, that's in the official specs.  And glancing at the code, it looks to me like the highest level test is run in the same way.

https://pkg.go.dev/testing#T.Run

> Run runs f as a subtest of t called name. It runs f in a separate goroutine and blocks until f returns or calls t.Parallel to become a parallel test. Run reports whether f succeeded (or at least did not fail before calling t.Parallel).
> 
> Run may be called simultaneously from multiple goroutines, but all such calls must return before the outer test function for t returns.
>